task/generate-labels/0.1/generate-labels.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: generate-labels
+spec:
+  description: |
+    Generate labels based on templates.
+
+    Usage may look like the following.
+
+    > - name: generate-labels
+    >   params:
+    >   - name: LABEL_TEMPLATES
+    >     value: '["release=$SOURCE_DATE_EPOCH", "build-date=$SOURCE_DATE"]'
+
+    The following environment variables are defined for use in LABEL_TEMPLATES
+
+    * ACTUAL_DATE - a date time string containing the time this task runs, formatted +'%Y-%m-%dT%H:%M:%SZ'
+    * ACTUAL_DATE_EPOCH - the timestamp at the time this task runs
+    * SOURCE_DATE - a date time string containing the provided source timestamp, formatted +'%Y-%m-%dT%H:%M:%SZ'
+    * SOURCE_DATE_EPOCH - the timestamp provided as a param meant to represent the timestamp at which the source was last modified
+
+  params:
+  - name: label-templates
+    description: A json array of templates that should be rendered and exposed as a list of labels
+    type: string
+  - name: source-date-epoch
+    description: A standardised environment variable for build tools to consume in order to produce reproducible output.
+    default: ""
+  results:
+    - name: labels
+      description: The rendered labels, rendered from the provided templates
+      type: array
+  steps:
+    - name: render
+      image: quay.io/konflux-ci/yq:latest@sha256:91a7fe73fd28bbfd93ebdcc68cdfd7087d4f2612c627ccd0d5f984190eb7dba6
+      env:
+      - name: LABEL_TEMPLATES
+        value: "$(params.label-templates)"
+      - name: SOURCE_DATE_EPOCH
+        value: "$(params.source-date-epoch)"
+      script: |
+          #!/bin/bash
+
+          ACTUAL_DATE_EPOCH=$(date -u +'%s')
+          ACTUAL_DATE=$(date -u --date=@"$ACTUAL_DATE_EPOCH" +'%Y-%m-%dT%H:%M:%SZ')
```

Comment:
In the example, you set `build-date` so this will result in the label being added twice: https://github.com/konflux-ci/build-definitions/blob/main/task/buildah-oci-ta/0.2/buildah-oci-ta.yaml#L379

If time-based labels are being set here, they will be inconsistent with the `build-date` label which is added in the buildah task itself. I know you deferred any reproducibility work to the future, but this seems like it would be likely to cause confusion since both are labels being set.

Reply:
Definitely.

Can we just remove `build-date` from the buildah task though? Others may be depending on it.

I guess, so long as enterprise contract doesn't require it - we could remove it and guide people to put this new task in place if and when they're ready.

Reply:
It can be possible for the removal to only affect the current users of the buildah tasks if we enable `generate-labels` by default and set it to define that build-date.

Or we could just check in the buildah task to see if the `build-date` label is being set externally and then not set it in the task.

We can consider fully removing the functionality in the next breaking change of the tasks.

Reply:
I think that would be good to do.

Reply:
Done in 100d2341.